tools/rostopic/src/rostopic/__init__.py

```diff
@@ -1145,25 +1161,46 @@ def eval_fn(m):
         parser.error("NUM_WIDTH must be an integer")
 
     field_filter_fn = create_field_filter(options.nostr, options.noarr)
+    value_transform_fn = create_value_transform(options.nostr, options.noarr)
     callback_echo = CallbackEcho(topic, None, plot=options.plot,
                                  filter_fn=filter_fn,
                                  echo_clear=options.clear, echo_all_topics=options.all_topics,
                                  offset_time=options.offset_time, count=msg_count,
-                                 field_filter_fn=field_filter_fn, fixed_numeric_width=fixed_numeric_width)
+                                 field_filter_fn=field_filter_fn, fixed_numeric_width=fixed_numeric_width,
+                                 value_transform_fn=value_transform_fn)
     try:
         _rostopic_echo(topic, callback_echo, bag_file=options.bag)
     except socket.error:
         sys.stderr.write("Network communication failed. Most likely failed to communicate with master.\n")
 
-def create_field_filter(echo_nostr, echo_noarr):
-    def field_filter(val):
+def create_value_transform(echo_nostr, echo_noarr):
+    def value_transform(val):
         fields = val.__slots__
         field_types = val._slot_types
-        for f, t in zip(val.__slots__, val._slot_types):
+        transformed = []
+        for index, (f, t) in enumerate(zip(fields, field_types)):
             if echo_noarr and '[' in t:
-                continue
+                f_val = val.__getattribute__(f)
+                transformed.append((index, t, f, f_val))
+                val.__setattr__(f, '<array type: %s, length: %s>' %
+                                (t.rstrip('[]'), len(f_val)))
+                val._slot_types[index] = 'string'
             elif echo_nostr and 'string' in t:
-                continue
+                f_val = val.__getattribute__(f)
+                transformed.append((index, t, f, f_val))
+                val.__setattr__(f, '<string length: %s>' % len(f_val))
+        def untransform_fn(val):
+            for index, type_, f, f_val in transformed:
+                val._slot_types[index] = type_
+                val.__setattr__(f, f_val)
+            return val
+        return val, untransform_fn
+    return value_transform
+
+def create_field_filter(echo_nostr, echo_noarr):
```

Comment:
I am not sure if this function is used anywhere outside by now it doesn't filter anything anymore.